secure_message/repository/modifier.py

```diff
@@ -97,3 +99,45 @@ def mark_message_as_read(message, user):
                 raise InternalServerError(description="Error adding read information to message")
         Modifier.remove_label(unread, message, user)
         return True
+
+    @staticmethod
+    def close_conversation(metadata, user):
+        bound_logger = logger.bind(conversation_id=metadata.id, user_id=user.user_uuid)
+        bound_logger.info("Getting user details")
+
+        user_details = internal_user_service.get_user_details(user.user_uuid)
+        bound_logger.info("Successfully retrieved user details")
+
+        try:
+            bound_logger.info("Closing conversation")
+            metadata.is_closed = True
+            metadata.closed_at = datetime.utcnow()
+            metadata.closed_by = f"{user_details.get('firstName')} {user_details.get('lastName')}"
+            metadata.closed_by_uuid = user.user_uuid
+            db.session.commit()
+        except SQLAlchemyError:
+            db.session.rollback()
+            logger.exception("Database error occured while closing conversation")
```

Comment:
the same here , you want to give more info that you can in the error.

Reply:
You actually caught something I hadn't noticed!  There was a bound logger already there that I wasn't using.  Good catch ^^